src/app/core/components/gift-storage/gift-storage.component.ts

```diff
@@ -0,0 +1,148 @@
+/* @format */
+import { Component, OnDestroy } from '@angular/core';
+import {
+  UntypedFormGroup,
+  UntypedFormBuilder,
+  Validators,
+  AbstractControl,
+  AsyncValidatorFn,
+  FormControl,
+  ValidationErrors,
+  ValidatorFn,
+} from '@angular/forms';
+import { AccountVO } from '@models/index';
+import { Observable, BehaviorSubject, Subscription, of, timer } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { Dialog } from '../../../dialog/dialog.service';
+import { AccountService } from '../../../shared/services/account/account.service';
+
+@Component({
+  selector: 'pr-gift-storage',
+  templateUrl: './gift-storage.component.html',
+  styleUrls: ['./gift-storage.component.scss'],
+})
+export class GiftStorageComponent implements OnDestroy {
+  giftForm: UntypedFormGroup;
+  availableSpace: string;
+  account: AccountVO;
+  bytesPerGigabyte = 1073741824;
+
+  public isSuccessful: boolean = false;
+  public giftResult: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(
+    false
+  );
+  private sub: Subscription;
+  isAsyncValidating: boolean;
+
+  constructor(
+    private fb: UntypedFormBuilder,
+    private accountService: AccountService,
+    private dialog: Dialog
+  ) {
+    this.account = this.accountService.getAccount();
+    this.availableSpace = this.bytesToGigabytes(this.account?.spaceLeft);
+    this.giftForm = this.fb.group({
+      email: [
+        '',
+        {
+          validators: [Validators.email, Validators.required],
+          asyncValidators: [this.emailValidator()],
+          updateOn: 'blur',
+        },
+      ],
+      amount: [
+        '',
+        {
+          validators: [
+            Validators.required,
+            Validators.min(0),
+            Validators.max(Number(this.availableSpace)),
+            this.integerValidator,
+          ],
+          updateOn: 'change',
+        },
+      ],
+      message: ['', []],
+    });
+    this.sub = this.giftResult.subscribe((isSuccessful) => {
+      if (isSuccessful) {
+        const giftedAmount = Number(this.giftForm.value.amount);
+        const remainingSpaceAfterGift =
+          Number(this.availableSpace) - giftedAmount;
+        this.availableSpace = String(remainingSpaceAfterGift);
+
+        const remainingSpaceInBytes =
+          remainingSpaceAfterGift * this.bytesPerGigabyte;
+
+        const newAccount = new AccountVO({
+          ...this.account,
+          spaceLeft: remainingSpaceInBytes,
+          spaceTotal:
+            this.account.spaceTotal - giftedAmount * this.bytesPerGigabyte,
+        });
+
+        this.accountService.setAccount(newAccount);
+        this.isSuccessful = isSuccessful;
+      }
```

Comment:
This whole function that runs when the Gifting Result is successful is also not covered by tests.  Do we want to test this bit of functionality?